holoviews/plotting/bokeh/chart.py

```diff
@@ -906,18 +904,38 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            try:
+                xdiff = np.diff(xvals)
+                if len(np.unique(xdiff)) == 1 and xdiff[0] == 0:
+                    xdiff = 1
+                if is_dt:
+                    width = xdiff.astype('timedelta64[ms]').astype(np.int32) * width
+                else:
+                    width = width / xdiff
+                width = 1 - np.repeat(np.min(np.abs(width)), len(xvals))
```

Comment:
Is this to have equidistant space between the values? 

I haven't followed the thread closely, so I assume this is what was agreed upon.

Reply:
It follows matplotlib convention at least. It'd be strange for x=1 to span from x=1-100
<img width="645" alt="image" src="https://github.com/holoviz/holoviews/assets/15331990/e4cfe96d-83ca-4a7a-aada-efd1ef029dd1">